cloudinit/sources/DataSourceVMware.py

```diff
@@ -53,6 +56,60 @@
 WAIT_ON_NETWORK_IPV4 = "ipv4"
 WAIT_ON_NETWORK_IPV6 = "ipv6"
 
+SUPPORTED_UPDATE_EVENTS_GUEST_INFO_KEY = "cloudinit.update-events.supported"
+ENABLED_UPDATE_EVENTS_GUEST_INFO_KEY = "cloudinit.update-events.enabled"
+
+# Support network reconfiguration for the following events:
+# TODO(akutz) Add support for METADATA_CHANGE and USER_REQUEST when
+#             those events are supported by Cloud-Init.
+SUPPORTED_UPDATE_EVENTS = {
+    EventScope.NETWORK: {
+        EventType.BOOT,
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.HOTPLUG,
+    }
+}
+
+# By default this datasource only configures the network when
+# booting a new instance or when a NIC is added or removed. If
+# a control plane like VM Service wishes the network to be
+# reconfigured on each boot, it should send in vendordata that
+# overrides the default update events.
+DEFAULT_UPDATE_EVENTS = {
+    EventScope.NETWORK: {
+        EventType.BOOT_NEW_INSTANCE,
+        EventType.HOTPLUG,
+    }
+}
+
+# Only trigger hook-hotplug on NICs with VMware drivers. Avoid
+# triggering the hook on Docker virtual NICs, etc. Please note, this
+# means the hotplug event will not work for SR-IOV NICs since their
+# driver ID is not in the list below. This is because an SR-IOV NIC
+# could present just about any device ID, and thus cannot be known in
```

Comment:
Reads a little weird.  Did you mean "could present itself as just about any device ID"?

Reply:
I think we're saying the same thing. I can try to make it more clear.